dbt_subprojects/daily_spellbook/models/xrpl/gas_xrpl_fees.sql

```diff
@@ -0,0 +1,169 @@
+        ,AVG(price) AS price_usd
+    FROM {{ source('prices', 'usd') }}
+    WHERE symbol = 'XRP' 
+        AND blockchain IS NULL
+    GROUP BY 1
+),
+xrpl_gas_fees AS (
+    SELECT 
+        'xrpl' AS blockchain
+        ,CAST(DATE_TRUNC('month', CAST(ledger_close_date AS TIMESTAMP)) AS DATE) AS block_month
+        ,CAST(DATE_TRUNC('day', CAST(ledger_close_date AS TIMESTAMP)) AS DATE) AS block_date
+        ,CAST(ledger_close_date AS TIMESTAMP) AS block_time
+        ,ledger_index AS block_number
+        ,CAST(hash AS VARCHAR) AS tx_hash
+        ,CAST(account AS VARCHAR) AS tx_from
+        ,CAST(destination AS VARCHAR) AS tx_to
+        ,CAST(NULL AS DOUBLE) AS gas_price
+        ,CAST(NULL AS DOUBLE) AS gas_used
+        ,CAST(NULL AS DOUBLE) AS gas_limit
+        ,CAST(NULL AS DOUBLE) AS gas_limit_usage
+        ,CAST(NULL AS DOUBLE) AS max_fee_per_gas
+        ,CAST(NULL AS DOUBLE) AS priority_fee_per_gas
+        ,CAST(NULL AS DOUBLE) AS max_priority_fee_per_gas
+        ,CAST(NULL AS DOUBLE) AS base_fee_per_gas
+        ,CAST(NULL AS DOUBLE) AS l1_fee
+        ,CAST(NULL AS DOUBLE) AS l1_gas_used
+        ,CAST(NULL AS DOUBLE) AS l1_gas_price
+        ,CAST(NULL AS DOUBLE) AS l1_fee_scalar
+        ,'XRP' AS currency_symbol
+        ,CAST(fee AS DOUBLE) / 1000000.0 AS tx_fee
+        ,CASE 
+            WHEN p.price_usd IS NOT NULL 
+            THEN (CAST(fee AS DOUBLE) / 1000000.0) * p.price_usd 
+            ELSE NULL 
+        END AS tx_fee_usd
+        ,CAST(fee AS VARCHAR) AS tx_fee_raw
+        ,ARRAY[
+            CAST(
+                ROW(
+                    'base_fee',
+                    CAST(fee AS DOUBLE) / 1000000.0
+                ) AS ROW(fee_type VARCHAR, amount DOUBLE)
+            )
+        ] AS tx_fee_breakdown
+        ,ARRAY[
+            CAST(
+                ROW(
+                    'base_fee',
+                    CASE 
+                        WHEN p.price_usd IS NOT NULL 
+                        THEN (CAST(fee AS DOUBLE) / 1000000.0) * p.price_usd 
+                        ELSE NULL 
+                    END
+                ) AS ROW(fee_type VARCHAR, amount DOUBLE)
+            )
+        ] AS tx_fee_breakdown_usd
+        ,ARRAY[
+            CAST(
+                ROW(
+                    'base_fee',
+                    CAST(fee AS VARCHAR)
+                ) AS ROW(fee_type VARCHAR, amount VARCHAR)
+            )
+        ] AS tx_fee_breakdown_raw
+        ,'XRP' AS tx_fee_currency
+        ,CAST(NULL AS VARCHAR) AS block_proposer
+        ,CASE 
+            WHEN JSON_EXTRACT_SCALAR(metadata, '$.TransactionIndex') IS NOT NULL THEN 
+                CAST(JSON_EXTRACT_SCALAR(metadata, '$.TransactionIndex') AS BIGINT)
+            ELSE NULL 
+        END AS tx_index
+        ,transaction_type AS tx_type
+        ,JSON_EXTRACT_SCALAR(metadata, '$.TransactionResult') AS transaction_result
+        ,sequence
+        ,hash || '_fee' AS unique_key
```

Comment:
does this unique key add any value downstream other than using in model config for uniqueness?
if not, i think we try to move away from it and follow EVM standard instead.
for EVM, unique keys in model config are just partition column + tx_hash, and that is enough to be unique. we should have one row per tx.